python/ray/workflow/step_executor.py

```diff
@@ -146,13 +148,33 @@ def execute_workflow(workflow: "Workflow") -> "WorkflowExecutionResult":
     if workflow.executed:
         return workflow.result
     workflow_data = workflow.data
-    baked_inputs = _BakedWorkflowInputs.from_workflow_inputs(
-        workflow_data.inputs)
+
+    # If the outer workflow step skips checkpointing, it would
+    # update the context of all inner steps (except the output step).
```

Comment:
Why not the output step? I thought we would also want to skip checkpointing for that step.